torch/onnx/symbolic_opset13.py

```diff
@@ -755,11 +755,14 @@ def repeat_interleave(
 @symbolic_helper.parse_args("v", "i", "i", "i")
 @_beartype.beartype
 def diagonal(g: jit_utils.GraphContext, self, offset, dim1, dim2):
+    rank = symbolic_helper._get_tensor_rank(self)
```

Comment:
what if ` rank is None`? when the model is exported with ` torch.jit.script`, ` self.type()` could be `None`, forcing `_get_tensor_rank` to return None too